modules/apps/layout/layout-admin-web/src/main/java/com/liferay/layout/admin/web/internal/info/item/provider/LayoutInfoItemLanguagesProvider.java

```diff
@@ -27,13 +41,99 @@
 	implements InfoItemLanguagesProvider<Layout> {
 
 	@Override
-	public String[] getAvailableLanguageIds(Layout layout) {
-		return layout.getAvailableLanguageIds();
+	public String[] getAvailableLanguageIds(Layout layout)
+		throws PortalException {
+
+		if (!layout.isTypeContent()) {
+			return layout.getAvailableLanguageIds();
+		}
+
+		Set<String> availableLocalesIds = new HashSet<>();
+
+		Set<Locale> siteAvailableLocales = _language.getAvailableLocales(
+			layout.getGroupId());
+
+		List<FragmentEntryLink> fragmentEntryLinks =
+			_fragmentEntryLinkLocalService.getFragmentEntryLinksByPlid(
+				layout.getGroupId(), layout.getPlid());
+
+		for (FragmentEntryLink fragmentEntryLink : fragmentEntryLinks) {
+			JSONObject editableValuesJSONObject =
+				JSONFactoryUtil.createJSONObject(
+					fragmentEntryLink.getEditableValues());
+
+			for (String translatableFragment : _TRANSLATABLE_FRAGMENTS) {
+				availableLocalesIds.addAll(
+					_getTranslatableFragmentLocaleIds(
+						editableValuesJSONObject, translatableFragment,
+						siteAvailableLocales));
+			}
+		}
+
+		return availableLocalesIds.toArray(new String[0]);
 	}
 
 	@Override
 	public String getDefaultLanguageId(Layout layout) {
 		return layout.getDefaultLanguageId();
 	}
 
+	private Set<String> _getTranslatableFragmentLocaleIds(
```

Comment:
Can you change "LocaleIds" to "LanguageIds"?

Reply:
![](https://media4.giphy.com/media/RLW9YEaSBfBMt79fm4/giphy.gif)